File: leetcode/array/maximum-subarray.cpp
class Solution {
public:
  int maxSubArray(vector<int> &nums) {
    int l = nums.size(), sum = nums[0], maxx = nums[0];

    for (int i = 1; i < l; i++) {
      sum += nums[i];
      if (sum < nums[i])
        sum = nums[i];
      if (maxx < sum)
        maxx = sum;
    }

    return maxx;
  }
};

File: leetcode/array/maximum-subarray.go
func maxSubArray(nums []int) int {
	l, s, m := len(nums), nums[0], nums[0]
	for i := 1; i < l; i++ {
		s += nums[i]
		if nums[i] > s {
			s = nums[i]
		}
		if s > m {
			m = s
		}
	}
	return m
}
